csrc/dynamic_transform.cpp

```diff
@@ -446,8 +540,10 @@ void DynamicTransformConcretizer::checkConcretizedUses(
 // concretized. Since symbolic IDs may be propagated down to
 // consumers, those domains need to be concretized accordingly.
 void DynamicTransformConcretizer::mutate(TensorView* tv) {
-  if (!tv->domain()->hasSymbolicAxis()) {
-    return;
+  for (auto root_id : tv->getRootDomain()) {
```

Comment:
This looks a bit confusing. A root ID may be mutated here, but there's also `propagateFromProducerToConsumer` below, which may also mutate a root ID. Is there any concern of conflicts?

Reply:
At line 659 in `propagateFromProducerToConsumer`, we reflect the earlier mutation by basing the new mutated ID on `maybeMutated(root_id)`. That way we compose both mutations.

Reply:
Well, what happens if the same root ID is mutated both at line 546 and within line 551? Doesn't the latter just overwrite the mutation at line 546?

Reply:
Oh, are you referring to line 659 before this PR?

Reply:
It will overwrite the mutation, but since we using `IterDomainBuilder(maybeMutated(root_id)->as<IterDomain>())` as the basis of the new mutation lets us update the `IterType` without changing other mutations like the extent.

Reply:
OK, I see it now. Maybe it'd be helpful to add that to the code comment too.